pkg/kubelet/allocation/allocation_manager.go

```diff
@@ -61,18 +64,56 @@ type Manager interface {
 
 	// RemoveOrphanedPods removes the stored state for any pods not included in the set of remaining pods.
 	RemoveOrphanedPods(remainingPods sets.Set[types.UID])
+
+	// Run starts the allocation manager. This is currently only used to handle periodic retry of
+	// pending resizes.
+	Run()
+
+	// PushPendingResize queues a pod with a pending resize request for later reevaluation.
+	// If a change in the resize queue is detected, pending resizes will be retried.
+	PushPendingResize(pod *v1.Pod)
+
+	// RetryPendingResizes signals the allocation manager to retry all pending resize requests.
+	RetryPendingResizes()
+
+	// HandlePodResourcesResize determines if a pod resize is feasible. If so,
+	// it updates the allocation and sets the pod resize conditions accordingly.
+	HandlePodResourcesResize(pod *v1.Pod) error
 }
 
 type manager struct {
 	allocated state.State
 	actuated  state.State
+
+	canResizePod   func(pod *v1.Pod) (bool, string, string)
+	triggerPodSync func(pod *v1.Pod)
+
+	statusManager       status.Manager
+	skipPeriodicRetries bool
+	pendingResizeChan   chan struct{}
+
+	pendingResizesLock     sync.RWMutex
+	podsWithPendingResizes []*v1.Pod
```

Comment:
You should either need a mutex or a channel for this, but probably not both. Either move all operations on `podsWithPendingResizes` into the loop triggered by the channel, or just call functions on it directly and guard it with a mutex.